test/Microsoft.ServiceFabric.Actors.Tests/Diagnostics/PerformanceCounterDiagnosticEventsTest.cs

```diff
@@ -0,0 +1,369 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Fabric;
+using System.Fabric.Common;
+using Fuzzy;
+using Inspector;
+using Microsoft.ServiceFabric.Actors.Runtime;
+using Microsoft.ServiceFabric.Actors.Tests;
+using Microsoft.ServiceFabric.Diagnostics;
+using Microsoft.ServiceFabric.Services.Remoting;
+using Moq;
+using Xunit;
+
+namespace Microsoft.ServiceFabric.Actors.Diagnostics
+{
+    public class PerformanceCounterDiagnosticEventsTest
+    {
+        readonly static IFuzz fuzzy = new RandomFuzz(Environment.TickCount);
+
+        readonly IDiagnosticEvents sut;
+
+        readonly IClock clock = Mock.Of<IClock>();
+
+        readonly static ActorTypeInformation actorTypeInfo = ActorTypeInformation.Get(typeof(TestActor));
+        readonly PerformanceCounterProviderV2 performanceCounterProvider = new PerformanceCounterProviderV2(Guid.NewGuid(), actorTypeInfo);
+
+        protected PerformanceCounterDiagnosticEventsTest() => sut = new PerformanceCounterDiagnosticEvents(performanceCounterProvider, clock);
+
+
+        public class Constructor : PerformanceCounterDiagnosticEventsTest
+        {
+            [Fact]
+            public void WithParametersSetsValue()
+            {
+                var providerField = sut.Field<PerformanceCounterProviderV2>().Value;
+                Assert.Equal(performanceCounterProvider, providerField);
+
+                var clockField = sut.Field<IClock>().Value;
+                Assert.Equal(clock, clockField);
+            }
+
+            [Fact]
+            public void ThrowsOnNullProvider()
+            {
+                var exception = Assert.Throws<ArgumentNullException>(() =>
+                {
```

Comment:
Could you simplify the lambdas? Even if you don't want to place it on a single line, there is no need or benefit to the curly braces.